src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -163,6 +163,7 @@ public JsonConverter GetConverter(Type typeToConvert)
 
             if (converter != null)
             {
+                Debug.Assert(converter.TypeToConvert != null);
```

Comment:
Why is this guaranteed?

Reply:
Not guaranteed, that is why i'm asserting here, i would bang it if it was guaranteed

Reply:
So, if `converter.TypeToConvert` was null, we could still null ref (in release)? To me, this look like something we should check and throw for rather than using a `Debug.Assert`. Although, I am not yet sure what test case would get us to exercise that code path.

```C#
Type? converterTypeToConvert = converter.TypeToConvert;

if (converterTypeToConvert == null || (!converterTypeToConvert.IsAssignableFrom(typeToConvert) &&
    !typeToConvert.IsAssignableFrom(converterTypeToConvert)))
{
    ThrowHelper.ThrowInvalidOperationException_SerializationConverterNotCompatible(converter.GetType(), typeToConvert);
}
```